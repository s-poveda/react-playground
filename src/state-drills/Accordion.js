import React, { Component } from 'react';

class Accordion extends Component {
	constructor(props) {
		super(props);
		this.state = { expandedKey: this.props.expandedKey ? this.props.expandedKey : null }
	}
	static defaultProps = {
		sections: []
	}

	onButtonClick = (key) => {
		if (this.state.expandedKey == key) return this.setState({ expandedKey: null });
		this.setState({ expandedKey: key })
	}

	generateLIs() {
		return this.props.sections.map((section, i) => {
			return (
				<li key={i}>
					<button onClick={() => this.onButtonClick(i)}>{section.title}</button>
					{this.state.expandedKey === i && <p>{section.content}</p>}
				</li>
			)
		});
	}

	render() {
		return (
			<div>
				<ul>
					{this.generateLIs()}
				</ul>
			</div>)
	}
}

export default Accordion;
