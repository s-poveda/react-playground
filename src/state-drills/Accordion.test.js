import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';
// enzyme
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
configure({ adapter: new Adapter() });



describe('Accordion componenet tests', () => {
  it('renders without crashing: smoke test', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion sections={[
      {
        title: 'First title',
        content: 'First Content'
      },
      {
        title: 'Second title',
        content: 'Second Content'
      },
      {
        title: 'Third title',
        content: 'Third Content'
      }
    ]} />, div)
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI without props supplies', () => {
    const tree = renderer
      .create(<Accordion />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the UI without crashing with click being passed', () => {
    const tree = renderer
      .create(<Accordion sections={[
        {
          title: 'First title',
          content: 'First Content'
        },
        {
          title: 'Second title',
          content: 'Second Content'
        },
        {
          title: 'Third title',
          content: 'Third Content'
        }
      ]}
        expandedKey={0}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the UI without crashing with props', () => {
    const tree = renderer
      .create(<Accordion sections={[
        {
          title: 'First title',
          content: 'First Content'
        },
        {
          title: 'Second title',
          content: 'Second Content'
        },
        {
          title: 'Third title',
          content: 'Third Content'
        }
      ]}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders buttons', () => {
    const wrapper = shallow(<Accordion sections={[{
      title: 'Third title',
      content: 'Third Content'
    }]} />)
    wrapper.find('button').simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})