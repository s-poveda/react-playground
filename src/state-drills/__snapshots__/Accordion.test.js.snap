// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion componenet tests renders buttons 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Third title
      </button>
      <p>
        Third Content
      </p>
    </li>
  </ul>
</div>
`;

exports[`Accordion componenet tests renders the UI without crashing with click being passed 1`] = `
<div>
  <ul>
    <li>
      <button
        onClick={[Function]}
      >
        First title
      </button>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Second title
      </button>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Third title
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion componenet tests renders the UI without crashing with props 1`] = `
<div>
  <ul>
    <li>
      <button
        onClick={[Function]}
      >
        First title
      </button>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Second title
      </button>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Third title
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion componenet tests renders the UI without props supplies 1`] = `
<div>
  <ul />
</div>
`;
